### ISSUE TITLE
plugin_geoip: Failed load GeoIP2 data (libmaxminddb 1.2.0)

### ISSUE THREAD

**Issue by kriegsmanj** [gdnsd/gdnsd#112]
When using libmaxminddb 1.2.0 and gdnsd 2.2.3

```# info: plugin_geoip: map '**': Loading GeoIP2 database '/etc/maxmind/databases/GeoIP2-City.mmdb': Version: 2.0, Type: GeoIP2-City, IPVersion: 6, Timestamp: 2016-03-29 04:14:27 UTC
*** stack smashing detected ***: gdnsd terminated```

- https://github.com/maxmind/libmaxminddb/releases/tag/1.2.0

**Comment by blblack** [comment]
Not surprising I guess, given the 1.1.5 fiasco (which gdnsd 2.2.3 fixed a day before the mm 1.2.0 release).  I guess for 1.2 we'll check for the new fields are configure/compile-time and have some ifdefs around new code...

**Comment by blblack** [comment]
https://github.com/blblack/gdnsd/tree/mmdb-120 has draft code to get 1.2.0 working for an eventual gdnsd-2.2.4 release.  I've tested it fairly extensively across mmdb versions with the existing testsuite, but haven't tried newer/commercial datasets, etc.  I'm not 100% happy with the version-compatibility barrier that's set at compile-time either, but I don't see any great alternatives.

**Comment by extremeshok** [comment]
just a thought.
what if you use your own db format and then have something to convert the maxmind or other geo databases into your format.

**Comment by blblack** [comment]
That's basically what's happening in memory at startup time, which is why we iterate the entire database instead of using the standard lookup functions.  The point behind our geoip code is to rewrite a new database that is optimized to the user's configuration, so that it can return optimal subnet masks for edns-subnet-client.

For example, a certain sub-space of IPv4 may be commonly split into /24's in MaxMind's database because each one is in a slightly different locale, but when the user's geoip location->datacenter mapping is applied for gdnsd purposes, it may turn out that 64 of them in a row (a whole /18) all map to the same datacenter in practice.  By returning the edns-client-subnet result with a /18 mask, we let DNS caches share a single response from the authserver with a broader range of clients.

**Comment by blblack** [comment]
Should be resolved as of 2.2.4 release
